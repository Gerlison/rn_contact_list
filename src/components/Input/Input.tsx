import React, { memo } from 'react';
import { TextInputProps } from 'react-native';
import styled from 'styled-components/native';

interface Props extends TextInputProps {}

const Input = (props: Props): JSX.Element => {
  return (
    <S.Container>
      <S.TextInput {...props} />
    </S.Container>
  );
};

const S = {
  Container: styled.View`
    width: 100%;
    margin-bottom: 24px;
    border-radius: 20px;
    border-color: #aeaeae;
    border-width: 1px;
  `,
  TextInput: styled.TextInput.attrs<Props>({
    placeholderTextColor: 'rgba(0,0,0,0.5)',
  })`
    padding: 20px;
    font-size: 14px;
    font-family: Poppins-Medium;
  `,
};

export default memo(Input);
