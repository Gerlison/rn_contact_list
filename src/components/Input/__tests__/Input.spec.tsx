import React from 'react';
import { render } from '@testing-library/react-native';

import Input from '../Input';

describe('Input', () => {
  it('SHOULD render correctly', () => {
    const component = render(<Input onChange={jest.fn()} value={'test'} />);
    expect(component).toBeDefined();
  });

  it('SHOULD match snapshot', () => {
    const component = render(
      <Input onChange={jest.fn()} value={'test'} />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
