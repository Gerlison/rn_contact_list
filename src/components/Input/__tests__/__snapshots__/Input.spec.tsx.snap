// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input SHOULD match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#aeaeae",
        "borderRadius": 20,
        "borderWidth": 1,
        "marginBottom": 24,
        "width": "100%",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    onChange={[MockFunction]}
    placeholderTextColor="rgba(0,0,0,0.5)"
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "fontFamily": "Poppins-Medium",
          "fontSize": 14,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="test"
  />
</View>
`;
